algorithms/polish.sbl

```diff
@@ -0,0 +1,240 @@
+      'e{s'}{c'}'    // infinitive, e.g. przynieść
+      'a{s'}{c'}'    // infinitive, e.g. popaść
+      'a{c'}'        // infinitive, e.g. czytać
+      'e{c'}'        // infinitive, e.g. chcieć
+      'i{c'}'        // infinitive, e.g. wozić
+      '{ak}{c'}'     // infinitive, e.g. marznąć
+      'a{l/}em'       'i{l/}em' 'em'  // past 1st person singular masculine, e.g. czytałem, chodziłem, marzłem
+      'a{l/}am'       'i{l/}am' 'am'  // past 1st person singular feminine, e.g. czytałam, chodziłam, marzłam
+      'a{l/}e{s'}'    'i{l/}e{s'}'    // past 2nd person singular masculine, e.g. czytałeś, chodziłeś
+      'a{l/}a{s'}'    'i{l/}a{s'}'    // past 2nd person singular feminine, e.g. czytałaś, chodziłaś
+      'a{l/}'         'i{l/}'         // past 3rd person singular masculine, e.g. czytał, chodził
+      'a{l/}a'        'i{l/}a'        // past 3rd person singular feminine, e.g. czytała, chodziła
+      'a{l/}o'        'i{l/}o'        // past 3rd person singular neuter, e.g. czytało, chodziło
+      'ali{s'}my'     'ili{s'}my'     // past 1st person plural virile, e.g. czytaliśmy, chodziliśmy
+      'a{l/}y{s'}my'  'i{l/}y{s'}my'  // past 1st person plural nonvirile, e.g. czytałyśmy, chodziłyśmy
+      'ali{s'}cie'    'ili{s'}cie'    // past 2nd person plural virile, e.g. czytaliście, chodziliście
+      'a{l/}y{s'}cie' 'i{l/}y{s'}cie' // past 2nd person plural nonvirile, e.g. czytałyście, chodziłyście
+      'ali'           'ili'           // past 3rd person plural virile, e.g. czytali, chodzili
+      'a{l/}y'        'i{l/}y'        // past 3rd person plural nonvirile, e.g. czytały, chodziły
+      '{ak}c'      // contemporary adverbial participle (transgressive), e.g. czytając
+      // different forms of active adjectival participle, e.g. czytający (one who is reading)
+      '{ak}cy'     // nominative singular (masculine)
+      '{ak}cego'   // genitive singular (masculine/neuter)
+      '{ak}cemu'   // dative singular (masculine)
+      '{ak}cym'    // instrumental/locative singular (masculine)
+      '{ak}ca'     // nominative singular (feminine)
+      '{ak}cej'    // genitive/dative/locative singular (feminine)
+      '{ak}c{ak}'  // accusative singular (feminine)
+      '{ak}ce'     // nominative/accusative singular (neuter)
+      '{ak}cych'   // genitive plural
+      '{ak}cymi'   // instrumental plural
+        (delete)
+      'aj'     // imperative, e.g. czytaj
+         (<- 'a')
+    )
+  )
+
+  define remove_adverbs_ends as (
+    setlimit tomark p1 for ([substring]) among (
+      'nie'  // e.g. dokładnie
+        (<- 'n')
+      'wie'  // e.g. możliwie
+        (<- 'w')
+      'rze'  // e.g. dobrze
+        (<- 'r')
+    )
+  )
+
+  define remove_plural_forms as (
+    setlimit tomark p1 for ([substring]) among (
+      '{o'}w'  // genitive plural, e.g. włos - włosów
+      'om'     // dative plural, e.g. włos - włosom
+      'ami'    // instrumental plural, e.g. włos - włosami
```

Comment:
_-ami_ actually gets removed from _włosami_ by `remove_nouns` (noted as "plural instrumental" there).  Maybe this is needed if _-ami_ can be exposed by removing another suffix but we also need to remove _-ami_ earlier to expose suffixes, but if so that kind of suggests the order of rules might not be quite right.